src/ChainlinkOracle.sol

```diff
@@ -51,36 +57,49 @@ contract ChainlinkOracle is IChainlinkOracle {
     /// - The vault's sample shares quoted as assets and the base feed prices don't overflow when multiplied.
     /// - The quote feed prices don't overflow when multiplied.
     /// - The vault, if set, is ERC4626-compliant.
-    /// @param vault Vault. Pass address zero to omit this parameter.
+    /// @param baseVault Base vault. Pass address zero to omit this parameter.
+    /// @param baseVaultConversionSample The sample amount of base vault shares used to convert to underlying.
+    /// Pass 1 if the base asset is not a vault. Should be chosen such that converting `baseVaultConversionSample` to assets has enough precision.
+    /// @param quoteVault Quote vault. Pass address zero to omit this parameter.
+    /// @param quoteVaultConversionSample The sample amount of quote vault shares used to convert to underlying.
+    /// Pass 1 if the base asset is not a vault. Should be chosen such that converting `quoteVaultConversionSample` to assets has enough precision.
     /// @param baseFeed1 First base feed. Pass address zero if the price = 1.
     /// @param baseFeed2 Second base feed. Pass address zero if the price = 1.
     /// @param quoteFeed1 First quote feed. Pass address zero if the price = 1.
     /// @param quoteFeed2 Second quote feed. Pass address zero if the price = 1.
-    /// @param vaultConversionSample The sample amount of vault shares used to convert to the underlying asset.
-    /// Pass 1 if the oracle does not use a vault. Should be chosen such that converting `vaultConversionSample` to
-    /// assets has enough precision.
     /// @param baseTokenDecimals Base token decimals.
     /// @param quoteTokenDecimals Quote token decimals.
     constructor(
-        IERC4626 vault,
+        IERC4626 baseVault,
+        uint256 baseVaultConversionSample,
+        IERC4626 quoteVault,
+        uint256 quoteVaultConversionSample,
         AggregatorV3Interface baseFeed1,
         AggregatorV3Interface baseFeed2,
         AggregatorV3Interface quoteFeed1,
         AggregatorV3Interface quoteFeed2,
-        uint256 vaultConversionSample,
         uint256 baseTokenDecimals,
         uint256 quoteTokenDecimals
     ) {
-        // The ERC4626 vault parameter is used to price `VAULT_CONVERSION_SAMPLE` of its shares, so it requires dividing
-        // by that number, hence the division by `VAULT_CONVERSION_SAMPLE` in the `SCALE_FACTOR` definition.
-        // Verify that vault = address(0) => vaultConversionSample = 1.
+        // The ERC4626 vaults parameter is used to price their respective conversion samples of their respective shares,
+        // so it requires multiplying by `QUOTE_VAULT_CONVERSION_SAMPLE` and dividing `BASE_VAULT_CONVERSION_SAMPLE` by
+        // the `SCALE_FACTOR` definition.
+        // Verify that vault = address(0) => vaultConversionSample = 1 for each vault.
```

Comment:
Because of this last comment I think we can reorder the requires to have VAULT_CONVERSION_SAMPLE_IS_NOT_ONE be next to each other